build-pipeline.yml

```diff
@@ -9,41 +9,13 @@
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
 parameters:
-  flake8_exclude: ""
+  # space separated list of Python packages that need to be pip installed.
+  extra_pip_dependencies: ""
 
-# This is a very simple build pipeline for now. It simply validates that the code
-# adheres to the Black code style and passes flake8.
+# This build pipeline will validate the code style and our test suite on
+# multiple platforms.
 jobs:
-- job: code_style_validation
-  displayName: Code Style Validation
-  pool:
-    vmImage: 'ubuntu-16.04'
-  steps:
-
-  # Use Python 3 for validating the code.
-  - task: UsePythonVersion@0
-    inputs:
-      versionSpec: '3.x'
-
-  # Install all the Python code validation packages.
-  - bash: python -m pip install black flake8
-    displayName: Install validators
-
```

Comment:
this got refactored into `code-style-validation.yml`. Note that the actual implementation just runs the pre-commit hook now. If the user was properly set up, then the pre-commit checks should pass.